pkg/ruler/remotequerier.go

```diff
@@ -297,6 +298,9 @@ func (q *RemoteQuerier) sendRequest(ctx context.Context, req *httpgrpc.HTTPReque
 		if err == nil {
 			return resp, nil
 		}
+		if code := grpcutil.ErrorToStatusCode(err); code/100 == 4 {
```

Comment:
I think we should also check the status code from `resp` because until we change configure the http grpc server configuration, 4xx errors will be returned as non errors.

Reply:
I have fixed this. Thank you